draft-irtf-cfrg-hpke.md

```diff
@@ -554,6 +554,42 @@ def Context.Open(aad, ct):
   return pt
 ~~~~~
 
+# Single-Shot APIs
+
+In many cases, applications encrypt only a single message to a recipient's public key. 
+This section provides HPKE APIs for "single-shot" encryption and decryption. These APIs
+build on the HPKE APIs specified in {{hpke-kem}} and {{hpke-dem}}. Each of them is built 
+on the following utility functions.
+
+~~~~~
+def ContextSeal(aad, pt, (enc, context)):
+  ct = context.Seal(aad, pt)
+  return enc, ct
+
+def ContextOpen(aad, ct, context):
+  return context.Open(aad, ct)
```

Comment:
I would still like this better if we just left it in a form that someone could fill in in the obvious way:

```
def SealMODE(pkR, info, aad, pt, ...):
  enc, ctx = SetupIMODE(pkR, info, ...)
  ct = ctx.sesal(aad, pt)
  return enc, ct
```

You can actually [implement things exactly this way](https://gist.github.com/bifurcation/4bb8a0444d5abe41fbe8a816a282b0fb) in C++: 

```
template<Mode M, typename... Args>
tuple<EncapsulatedKey, EncryptionContext>
setupI(const PublicKey& pkR, const Bytes& info, const Args&... args);

// ... instantiations ...

template<Mode M, typename... Args>
tuple<EncapsulatedKey, Bytes>
seal(const PublicKey& pkR,
     const Bytes& info,
     const Bytes& aad,
     const Bytes& pt,
     const Args&... args)
{
  auto [enc, ctx] = setupI<M>(pkR, info, args...);
  auto ct = ctx.seal(aad, pt);
  return { enc, ct };
}
```

Reply:
Good idea! I refactored it a bit based on this suggestion. Please have a look.